Feature/prebid

viacheslavbocharov opened the pull request:
Prebid ads module:

Enabled via import "virtual:plugins" and VITE_ENABLE_ADS.
Two horizontal slots: before and after <main> on all pages.
Two bidders: adtelligent (aid=350975) and bidmatic (source=886409).
/ads-debug page with live logs (auctionInit, bidResponse, bidWon, auctionEnd, renderError).
Render on bidResponse; in prod prefer pbjs.getHighestCpmBids() after auction end.

viacheslavbocharov committed:
feat(ads): first banner render via Prebid.js

viacheslavbocharov committed:
feat(ads):  banner render withou anchor

viacheslavbocharov committed:
feat(ads): horizontal slots above/below <main> with no scrollbars

viacheslavbocharov committed:
feat(ads): add bidmatic

viacheslavbocharov committed:
feat(ads): robust render error handling + debug event

- replace empty catch around pb.renderAd with typed error handling
- emit "renderError" custom event for /ads-debug stream
- log context (adUnitCode, adId, bidder, cpm, message) to console
- gracefully collapse failed slot (hide iframe, reset minHeight) to avoid blank space
- keep strict typings (no `any`)

Gunnar97 commented on src/app/router.tsx:
```diff
@@ -13,6 +14,7 @@ export const router = createBrowserRouter([
       { path: "/", element: <Home /> },
       { path: "/login", element: <Login /> },
       { path: "/register", element: <Register /> },
+      { path: "/ads-debug", element: <AdsDebug /> },
```
если нет pbjs страницы нет

viacheslavbocharov replied:
Fixed

Gunnar97 commented on src/modules/prebid/prebid.ts:
```diff
@@ -0,0 +1,228 @@
+    }
+  });
+
+  pb.onEvent("bidWon", (bid) => {
```
должно быть pbjs

viacheslavbocharov replied:
Fixed

Gunnar97 commented on vite.config.ts:
```diff
@@ -32,42 +33,92 @@ function virtualFlags() {
     },
-  },
-  server: { host: "localhost", port: 5174 },
+    server: { host: "localhost", port: 5174 },
```
переменные

viacheslavbocharov replied:
Fixed

viacheslavbocharov committed:
chore: gate /ads-debug by env, unify pbjs naming, env-driven dev server

- /ads-debug route only when VITE_ENABLE_ADS=true
- rename event bindings to `pbjs` for consistency
- read Vite host/port from env (VITE_DEV_HOST/VITE_DEV_PORT)
